src/data_conn.c

```diff
@@ -226,6 +226,14 @@ handle_data_conn_error(replica_t *r)
 	else if (r->state == ZVOL_STATUS_DEGRADED)
 		spec->degraded_rcount--;
 
+	if (spec->master_replica == r) {
```

Comment:
can this be moved to the end of handle_mgmt_conn_error?

Reply:
mgmt_conn is the one on which rebuild commands are sent, and, handle_mgmt_conn_error tears down the replica completely. So, it would be good to move to that fn.